src/utils.rs

```diff
@@ -208,4 +245,61 @@ mod test {
             assert_eq!(x, vec!(1u8; i));
         }
     }
+
+    #[test]
+    fn test_bin2hex() {
+        use std::str;
+
+        for (i, j) in vec![(0, "00\u{0}"), (1, "01\u{0}"), (10, "0a\u{0}"), (15, "0f\u{0}")] {
```

Comment:
Not a huge fan of trailing null byte, but that's how upstream does it and I'm not sure if we should change it.